### ISSUE TITLE
Reconsider over-eager warning

### ISSUE THREAD

**Issue by effigies** [grabbles/grabbit#76]
I suspect this is due to a refactoring in which directories, and not domains, are the objects iterated over, but this warning is too-easily triggered:

https://github.com/grabbles/grabbit/blob/a4eb51881e7ef8abb72a6db35f0541b1f06e63f6/grabbit/core.py#L446-L450

For example:

```Python
layout = gb.BIDSLayout([(bids_dir, 'bids'), (preproc_dir, ['bids', 'derivatives'])])
```

Because the `'bids'` domain appears twice, this is warning is displayed. However, I expect this to be very common among multi-root use cases, so we should reconsider this warning.

I see three possibilities:

1) Re-think the potentially conflicting behavior, given the new invocation patterns, and trigger in that case.
2) Do some work to predict whether an actual conflict may arise, and trigger only if that is the case.
3) Remove the warning because there are not actual conflicts.

**Comment by tyarkoni** [comment]
Agreed on both the diagnosis and cause. I think at least in the short-term, we can probably go with your third option. I was probably overly cautious, and I don't think one will run into this situation in the real world very often (plus, when one does run into it, in most cases the two domains will be identical anyway, so nothing bad should happen). I'd be fine commenting it out and maybe leaving a comment as a TODO somewhere.

**Comment by effigies** [comment]
I can do this next time I'm digging around in the code, but it's not a big enough deal for right now. Could be a good chance for a new contributor to get their feet wet, though.
